public/app/plugins/panel/geomap/utils/tooltip.ts

```diff
@@ -12,7 +13,7 @@ import { getMapLayerState } from './layers';
 export const setTooltipListeners = (panel: GeomapPanel) => {
   // Tooltip listener
   panel.map?.on('singleclick', panel.pointerClickListener);
-  panel.map?.on('pointermove', panel.pointerMoveListener);
+  panel.map?.on('pointermove', debounce(panel.pointerMoveListener, 200));
```

Comment:
I played around with multiple debounce delays and 200 seems to be the sweet spot where normal tooltip hover behavior isn't too painfully slow and the freezing / major lag is no longer an issue